host/tests/hdmiCEC_L3_Tests/hdmiCEC_test01_TransmitCECCommands.py

```diff
@@ -0,0 +1,152 @@
+# *
+# * If not stated otherwise in this file or this component's LICENSE file the
+# * following copyright and licenses apply:
+# *
+# * Copyright 2024 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# *
+# http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import os
+import sys
+import time
+
+# Append the current and parent directory paths to sys.path for module imports
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path))
+sys.path.append(os.path.join(dir_path, "../"))
+
+# Importing helper classes and modules for HDMI-CEC testing and logging
+from hdmiCECHelperClass import hdmiCECHelperClass
+from raft.framework.core.logModule import logModule
+
+class hdmiCEC_test01_TransmitCECCommands(hdmiCECHelperClass):
+    """
+    A class for testing HDMI-CEC functionality by transmitting and verifying CEC commands.
+    """
+    def __init__(self, log:logModule=None):
+        """
+        Initializes the test class with test-specific configurations.
+
+        Args:
+            log (logModule): Logging module instance to record test execution.
+        """
+        # Class variables
+        self.testName  = "test01_TransmitCECCommands"
+        self.qcID = '1'
+        self.broadcastAddress = 'f'
+
+        # Initialize the parent class
+        super().__init__(self.testName, self.qcID, log)
+
+    def testVerifyTxResults(self, txResult:str, rxResult:bool):
+        """
+        Verifies the tx and rx results.
+
+        Args:
+            txResult(str): tx result
+            rxResult(bool): rx result
+        Returns:
+            bool: Final result of the test.
+        """
+
+        return rxResult
+
+    def testFunction(self):
+        """
+        Main test function for transmitting HDMI-CEC commands and validating responses.
+
+        Steps:
+        1. Initialize HDMI-CEC module.
+        2. Add a logical address to the test device.
+        3. Get the logical address of the device.
+        4. List all connected CEC devices.
+        5. Iterate through devices and send commands to appropriate logical addresses.
+        6. Validate transmission and record results.
+        7. Clean up by removing logical addresses and terminating the HDMI-CEC module.
+
+        Returns:
+            bool: Final result of the test execution.
+        """
+
+        # Initialize the hdmiCEC module
+        self.testhdmiCEC.initialise()
```

Comment:
What if Init fails?  We are not validating this.  This is required to simulate a condition in source device where we try to connect more devices of same kind to check that init fails.